File: task3.js
/* Task 3 */

const STR_S = { one: 'секунда', some: 'секунды', many: 'секунд' };
const STR_M = { one: 'минута', some: 'минуты', many: 'минут' };
const STR_H = { one: 'час', some: 'часа', many: 'часов' };

var t1 = Number(process.argv[2]);
var t2 = Number(process.argv[3]);

var time = t1 + t2;
var timeS = time % 60;
var timeM = Math.floor(time / 60) % 60;
var timeH = Math.floor(time / 3600);

process.stdout.write((timeH ? numToString(timeH, STR_H) + ' ' : '') +
                     (timeM ? numToString(timeM, STR_M) + ' ' : '') +
                              numToString(timeS, STR_S) + '\n');
process.exit(0);

/**/
function numToString(num, temp) {
  mod = num % 10;
  return num + ' ' + (num > 10 && num < 20 || mod == 0 || mod >= 5 ?
                      temp.many : (mod == 1 ? temp.one : temp.some));
}

File: task3.rb
# Task 3

STR_K = [ 'one', 'some', 'many' ]
STR_S = Hash[*[STR_K, [ 'секунда', 'секунды', 'секунд' ]].transpose.flatten]
STR_M = Hash[*[STR_K, [ 'минута', 'минуты', 'минут' ]].transpose.flatten]
STR_H = Hash[*[STR_K, [ 'час', 'часа', 'часов' ]].transpose.flatten]

def num_to_string(num, temp)
  mod = num % 10
  return num.to_s + ' ' + (num > 10 && num < 20 || mod == 0 || mod >= 5 ?
                           temp['many'] : (mod == 1 ? temp['one'] : temp['some']))
end

t1 = ARGV[0].to_i
t2 = ARGV[1].to_i

time = t1 + t2
time_s = time % 60
time_m = (time / 60).floor % 60
time_h = (time / 3600).floor

puts (time_h.nonzero? ? num_to_string(time_h, STR_H) + ' ' : '') +
     (time_m.nonzero? ? num_to_string(time_m, STR_M) + ' ' : '') +
                        num_to_string(time_s, STR_S)

